### PR TITLE
[PATCH 2.0 v6] linux-gen: buffer: remove burst metadata from odp_buffer_hdr_t

### PR DESCRIPTION
**Pull request by kevinwangsk**
The total packet meta data size was 7 cache lines for Linux-generic.
With this change the size of the meta data is 4 cache lines for
Linux-generic.
Make scalable scheduler as the default scheduler.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -66,9 +66,6 @@ cache:
 env:
         - CONF=""
         - CONF="--disable-abi-compat"
-        - CONF="--enable-schedule-sp"
-        - CONF="--enable-schedule-iquery"
-        - CONF="--enable-schedule-scalable"
         - CONF="--enable-dpdk-zero-copy"
```
Comment (Bill-Fischofer-Linaro) [inline .travis.yml]: What do these changes have to do with the subject of this PR? Changes to Travis should at minimum be a separate commit (these are not "linux-generic: buffer" changes) and should probably be a separate PR with its own rationale.

File: platform/linux-generic/Makefile.am [hunk 2]
```diff
@@ -257,16 +257,9 @@ __LIB__libodp_linux_la_SOURCES = \
...
-			   queue/generic.c \
-			   queue/scalable.c \
 			   odp_rwlock.c \
 			   odp_rwlock_recursive.c \
 			   odp_schedule_if.c \
-			   schedule/generic.c \
-			   schedule/iquery.c \
-			   schedule/scalable.c \
-			   schedule/scalable_ordered.c \
-			   schedule/sp.c \
```
Comment (Bill-Fischofer-Linaro) [inline platform/linux-generic/Makefile.am]: Again it's not obvious how these are buffer changes.

File: platform/linux-generic/Makefile.am [hunk 3]
```diff
@@ -354,20 +347,31 @@ endif
...
+endif
 endif
+endif
+
 if ODP_SCHEDULE_SCALABLE
+__LIB__libodp_linux_la_SOURCES += queue/scalable.c
 queue/scalable.lo: AM_CFLAGS += -DIM_ACTIVE_MODULE
 else
+__LIB__libodp_linux_la_SOURCES += queue/generic.c
 queue/generic.lo: AM_CFLAGS += -DIM_ACTIVE_MODULE
```
Comment (Bill-Fischofer-Linaro) [inline platform/linux-generic/Makefile.am]: Same comment as above. Not relevant to this PR's subject matter.

File: platform/linux-generic/m4/odp_schedule.m4 [hunk 4]
```diff
@@ -15,12 +19,3 @@ AC_ARG_ENABLE([schedule-iquery],
...
 AM_CONDITIONAL([ODP_SCHEDULE_IQUERY], [test x$schedule_iquery_enabled = xyes])
-
-AC_ARG_ENABLE([schedule_scalable],
-    [  --enable-schedule-scalable   enable scalable scheduler],
-    [if test x$enableval = xyes; then
-	schedule_scalable_enabled=yes
-	AC_DEFINE([ODP_SCHEDULE_SCALABLE], [1],
-		  [Define to 1 to enable scalable scheduler])
-    fi])
-AM_CONDITIONAL([ODP_SCHEDULE_SCALABLE], [test x$schedule_scalable_enabled = xyes])
```
Comment (Bill-Fischofer-Linaro) [inline platform/linux-generic/m4/odp_schedule.m4]: Did to PRs somehow get accidentally merged? These have to be separated.

File: .travis.yml (hunk 1 above)
Comment (kevinwangsk) [reply .travis.yml]: Agree that. Travis changes should be in a separate commit.

File: platform/linux-generic/Makefile.am (hunk 2 above)
Comment (kevinwangsk) [reply platform/linux-generic/Makefile.am]: Agree. The makefile changes should be a separate commit as well.

**Comment by Bill-Fischofer-Linaro** [comment]
@kevinwangsk As we agreed during today's call, please rework this so that the buffer header fields are conditionally included when schedulers that currently reference them are used. That way all current schedulers will still compile and run and these fields are only omitted when the scalable scheduler is used. This will permit benchmarking to demonstrate the performance improvements of these changes. 

Actual field removal can be done at a later date when everyone is comfortable that the other schedulers currently using them are no longer needed, or they've been reworked to no longer use them.

**Comment by kevinwangsk** [comment]
Agree with that. I think previously patch #176 based on cloud-dev is enough at the current stage. Will move it to 2.0 branch.

**Comment by kevinwangsk** [comment]
Rebase it on the latest commit.

**Comment by heyi-linaro** [comment]
Merged and close this PR.
